cw721-timelock/src/msg.rs

```diff
@@ -0,0 +1,64 @@
+use cosmwasm_std::Addr;
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use andromeda_std::{
+    andr_exec, andr_instantiate, andr_query,
+    amp::{AndrAddr, Recipient},
+    common::{milliseconds::MillisecondsDuration},
+};
+use cw721::Cw721ReceiveMsg;
+
+#[andr_instantiate]
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub authorized_token_addresses: Option<Vec<AndrAddr>>,
+}
+
+#[andr_exec]
+#[cw_serde]
+pub enum ExecuteMsg {
+    ReceiveNft(Cw721ReceiveMsg),
+    ClaimNft {
+        cw721_contract: AndrAddr,
+        token_id: String,
+    },
+}
+
+#[cw_serde]
+pub enum Cw721HookMsg {
+    TimelockNft {
+        lock_duration: MillisecondsDuration,
+        recipient: Recipient,
+    }
+}
+
+#[andr_query]
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
```

Comment:
Would be useful to have an `IsLocked` query that returns a `bool`

Reply:
Thanks for your suggestion.
I solved this case.